packages/edit-site/src/components/block-editor/index.js

```diff
@@ -12,24 +18,136 @@ import {
 	WritingFlow,
 	ObserveTyping,
 	BlockList,
+	__experimentalUseResizeCanvas as useResizeCanvas,
+	useBlockSelectionClearer,
 } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import NavigateToLink from '../navigate-to-link';
 import { SidebarInspectorFill } from '../sidebar';
 
+const alignmentFix = {
+	css:
+		'.wp-block[data-align="full"],.wp-block.alignfull{max-width:100vw!important;width:100vw!important;}',
+};
+
+function IframeContent( { children, doc, head, styles, bodyClassName } ) {
+	const onFocus = useBlockSelectionClearer();
+
+	useEffect( () => {
+		doc.body.tabIndex = '-1';
+		doc.body.addEventListener( 'focus', onFocus );
+
+		return () => {
+			doc.body.removeEventListener( 'focus', onFocus );
+		};
+	}, [ onFocus ] );
+
+	useEffect( () => {
+		doc.body.className = bodyClassName;
+		doc.body.style.margin = '0';
```

Comment:
Why do we need this margin for? The reason I noticed is because [I'm using](https://github.com/nosolosw/global-styles-theme/blob/master/style.css#L16) the `margin: auto` trick to center content in the middle for test purposes and this body.margin messes up with it (note how tho content is not centered).

![Screenshot from 2020-10-06 16-34-45](https://user-images.githubusercontent.com/583546/95217151-28546900-07f3-11eb-86a7-4fcb41470b44.png)

Reply:
I think we can remove it now. It was meant as a reset when I first started working on this, but themes should reset this.

Reply:
I'm really starting to think that using margins like this to handle default "alignment"/width in a theme was a bad idea. It hinders blocks from defining their own left/right margins on their root element (something I ran into while working on #21711), interferes with inner-block styles (I ran into this while working on #23222), and in general feels really fragile. (#25892, for example, currently has the Quote block up against the left side of the screen, similar to the situation in the screenshot @nosolosw posted.)

I wonder if we should consider making a recommendation against this styling method and avoid it in future default WP themes.

Reply:
I remove the inline style. Does it fix the issue, @nosolosw?